client/my-sites/plans-grid/index.tsx

```diff
@@ -106,6 +108,10 @@ const WrappedComparisonGrid = ( {
 		onUpgradeClick,
 	} );
 
+	const getFreeTrialSlugForPlan = useGetFreeTrialSlugForPlan( {
+		flowName: flowName ?? null,
+	} );
```

Comment:
I think this adds unnecessary clutter, depends on `flowName` that we want to steer away from (we are hoping to completely uncouple grid from this concept), and also breaks away from the data foundation used -> all of the relevant metadata for grid plans is provided through the `GridPlan` type.

Have we considered feeding the free trial slugs through [use-grid-plans](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans.tsx) hook? So I would have expected a hook to be called from there at the top level, then grab the available trial slugs and attach to the grid plans/objects returned, just like we do for the other metadata. 

maybe take a look at how `usePricingMetaForGridPlans` or `usePricedAPIPlans` are used and follow along?

but was there some other reason we chose to propagate this callback in the components instead? 🤔

Reply:
> maybe take a look at how `usePricingMetaForGridPlans` or `usePricedAPIPlans` are used and follow along?

and also try and use `intent` (in combination with `isInSignup` if you want to target just signup) instead of `flowName`.

Reply:
>but was there some other reason we chose to propagate this callback in the components instead? 🤔

Sheer ignorance 😁 Thanks for the help, I'll try and rewrite the code!